src/Files.App/Interacts/BaseLayoutCommandImplementationModel.cs

```diff
@@ -576,52 +577,120 @@ public async Task CreateFolderWithSelection(RoutedEventArgs e)
 
 		public async Task CompressIntoArchive()
 		{
-			string archivePath;
+			var (sources, directory, fileName) = GetCompressDestination();
+
+			var dialog = new CreateArchiveDialog
+			{
+				FileName = fileName,
+			};
+			await dialog.ShowAsync();
+
+			if (!dialog.CanCreate)
+				return;
+
+			IArchiveCreator creator = new ArchiveCreator
+			{
+				Sources = sources,
+				Directory = directory,
+				FileName = dialog.FileName,
+				Password = dialog.Password,
+				FileFormat = dialog.FileFormat,
+				CompressionLevel = dialog.CompressionLevel,
+				SplittingSize = dialog.SplittingSize,
+			};
+
+			await CompressArchiveAsync(creator);
+		}
+
+		public async Task CompressIntoZip()
+		{
+			var (sources, directory, fileName) = GetCompressDestination();
+
+			IArchiveCreator creator = new ArchiveCreator
+			{
+				Sources = sources,
+				Directory = directory,
+				FileName = fileName,
+				FileFormat = ArchiveFormats.Zip,
+			};
+
+			await CompressArchiveAsync(creator);
+		}
+
+		public async Task CompressIntoSevenZip()
+		{
+			var (sources, directory, fileName) = GetCompressDestination();
+
+			IArchiveCreator creator = new ArchiveCreator
+			{
+				Sources = sources,
+				Directory = directory,
+				FileName = fileName,
+				FileFormat = ArchiveFormats.SevenZip,
+			};
+
+			await CompressArchiveAsync(creator);
+		}
+
+		private (string[] Sources, string directory, string fileName) GetCompressDestination()
+		{
 			string[] sources = associatedInstance.SlimContentPage.SelectedItems
 				.Select(item => item.ItemPath)
 				.ToArray();
 
-			if (sources.Length == 1)
-				archivePath = sources[0] + ".zip";
-			else
-			{
-				DynamicDialog archiveDialog = DynamicDialogFactory.GetFor_RenameDialog();
-				await archiveDialog.ShowAsync();
-				if (archiveDialog.DynamicResult != DynamicDialogResult.Primary)
-					return;
-				archivePath = Path.Combine(
-					associatedInstance.FilesystemViewModel.WorkingDirectory,
-					$"{(string)archiveDialog.ViewModel.AdditionalData}.zip");
-			}
+			if (sources.Length is 0)
+				return (sources, string.Empty, string.Empty);
+
+			string directory = associatedInstance.FilesystemViewModel.WorkingDirectory.Normalize();
+			string fileName = Path.GetFileName(sources.Length is 1 ? sources[0] : directory);
```

Comment:
nit:

```suggestion
			string fileName = Path.GetFileName(sources is [source] ? source : directory);
```

Reply:
This does not compile as the feature is in preview. Maybe I need to activate something.